crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -76,17 +78,60 @@ impl DynamicScene {
         scene
     }
 
+    /// Create a new dynamic scene from a given [`World`] at a given [`Entity`].
+    ///
+    /// All descendants entities from the given one will be extracted.
+    pub fn from_world_at_root(
+        world: &World,
+        root: Entity,
+        type_registry: &AppTypeRegistry,
+    ) -> Self {
+        let mut scene = DynamicScene::default();
+        let type_registry = type_registry.read();
+
+        // Extract all entities that are a descendant of root
+        let mut entities_of_interest = list_all_descendants(world, root);
+        entities_of_interest.push(root);
```

Comment:
Nit: This is more of a stylistic thing I suppose, but should the root come first in this list?
```suggestion
        let mut entities_of_interest =
            std::iter::once(root).chain(list_all_descendants(world, root));
```